.github/triage/tests/test_triage_history_bisection.py

```diff
@@ -0,0 +1,199 @@
+        git_cmd("cherry-pick", f1)
+        failing_nonlinear = git_cmd("rev-parse", "HEAD")
+        git_cmd("checkout", "main")
+
+        # yield all the info
+        yield {
+            "paths": {
+                "repo": repo_path,
+                "output": output_path,
+                "scripts": mock_scripts_path,
+            },
+            "commits": {
+                "good_main": m2,
+                "bad_main": m3,
+                "feature": f1,
+                "passing_nonlinear": passing_nonlinear,
+                "failing_nonlinear": failing_nonlinear,
+            },
+        }
+
+
+@pytest.mark.parametrize(
+    "scenario, passing_commit_key, failing_commit_key, expected_good_key, expected_bad_key",
+    [
+        (
+            "Non-Linear History",
+            "passing_nonlinear",
+            "failing_nonlinear",
+            "good_main",
+            "bad_main",
+        ),
+        ("Linear History", "good_main", "bad_main", "good_main", "bad_main"),
+    ],
+)
+def test_triage_scenarios(
+    triage_env,
+    monkeypatch,
+    scenario,
+    passing_commit_key,
+    failing_commit_key,
+    expected_good_key,
+    expected_bad_key,
+):
+    """Test the get_commit_history for linear and non-linear histories."""
+    paths = triage_env["paths"]
+    all_commits = triage_env["commits"]
+    jax_repo_path = paths["repo"] / "jax"
+    passing_versions = {"jax": all_commits[passing_commit_key]}
+    failing_versions = {"jax": all_commits[failing_commit_key]}
+    passing_versions_str = f"jax:{all_commits[passing_commit_key]}"
+    failing_versions_str = f"jax:{all_commits[failing_commit_key]}"
+
+    bazel_cache_path = (paths["output"] / "bazel-cache").resolve()
+    bazel_cache_path.mkdir()
+
+    arg_list = [
+        "--main-branch",
+        "main",
+        "--output-prefix",
+        str(paths["output"]),
+        "--container-runtime",
+        "local",
+        "--passing-versions",
+        passing_versions_str,
+        "--failing-versions",
+        failing_versions_str,
+        "--bazel-cache",
+        str(bazel_cache_path),
+        "--",
+        str(paths["scripts"] / "test-case.sh"),
+        str(jax_repo_path),
+        all_commits["bad_main"],
+    ]
+    args = parse_args(arg_list)
+    logger = logging.getLogger(f"Scenario-{scenario}")
+    logging.basicConfig(level=logging.INFO)
+    # mp to mock bazel
+    original_path = os.environ.get("PATH", "")
+    monkeypatch.setenv("PATH", f"{paths['scripts']}:{original_path}")
+
+    tool = TriageTool(args, logger)
+    tool.package_dirs = {"jax": str(jax_repo_path)}
+    tool.dynamic_packages = {"jax"}
+    tool.bisection_url = "local"
+    if scenario == "Linear History":
+        linear_build_script_path = paths["scripts"] / "build-jax.sh"
+        linear_build_script_path.write_text(
+            "#!/bin/sh\necho 'Mock linear build successful.'\nexit 0"
+        )
+        os.chmod(linear_build_script_path, 0o755)
```

Comment:
How about instead of doing this you just add an extra couple of commits on top of `F1` (or a cherry-pick of it) and use those for the linear case? i.e. `F1 -> L1 -> L2 -> L3` and the linear triage is between L1 and L3.